src/app.ts

```diff
@@ -138,12 +138,16 @@ if (PERF_LOGGING_ONLY) {
 }
 
 if (ENABLE_PERF_LOGGING) {
-  // @ts-expect-error
-  if (!global.nativePerformanceNow && window?.performance) {
-    // @ts-expect-error
-    global.nativePerformanceNow = () => window.performance.now()
+  const nativePerformanceNow = (): number => {
+    // @ts-expect-error - this is a hack to get around the fact that window is not typed
+    return window.performance.now()
```

Comment:
Take all this performance stuff, and move it to its own file. Do not attach it to the global scope, but instead `export const pnow =` and so forth for each performance function. If we want to do performance analysis in the future, we can just `import { pstart, pend, ... } from '../../perf.ts'` or whatever you call the new file, rather than grabbing it from the global scope.

Reply:
The reason why this is in global scope was for the convenience in putting it in any repo like login-ui. Admittedly since most of our code is in the webview, this doesn't help much although it would be nice to have some equivalent in there as well.